pg_search/src/postgres/storage/fsm.rs

```diff
@@ -958,20 +958,27 @@ pub mod v2 {
             when_recyclable: pg_sys::FullTransactionId,
             extend_with: impl Iterator<Item = pg_sys::BlockNumber>,
         ) {
+            let mut extend_with = extend_with.peekable();
```

Comment:
```suggestion
            // Must peek first because CREATE INDEX hashing uses the first block number
            let mut extend_with = extend_with.peekable();
```